crates/eldenring/src/cs/cs_rand.rs

```diff
@@ -0,0 +1,208 @@
+use shared::OwnedPtr;
+use vtable_rs::VPtr;
+
+#[vtable_rs::vtable]
+pub trait CSRandVmt {
+    fn destructor(&mut self, should_free: bool);
+    fn next_uint(&mut self) -> u32;
+    fn next_long(&mut self) -> u64;
+}
+
+pub struct CSRand<T: CSRandVmt> {
+    pub vftable: VPtr<dyn CSRandVmt, T>,
+}
+
+impl<T: CSRandVmt> CSRandVmt for CSRand<T> {
+    extern "C" fn destructor(&mut self, _should_free: bool) {}
+    extern "C" fn next_uint(&mut self) -> u32 {
+        0
+    }
+    extern "C" fn next_long(&mut self) -> u64 {
+        0
+    }
+}
```

Comment:
I'm probably missing something obvious, but these are game types, right? Why implement your own Rust vtables for them?